src/cls/rgw/cls_rgw.cc

```diff
@@ -2358,6 +2533,10 @@ int rgw_dir_suggest_changes(cls_method_context_t hctx,
             return ret;
           }
         }
+        if (header.resharding_in_logrecord()) {
+          bufferlist empty;
+          return reshard_log_index_operation(hctx, cur_change_key, cur_change.key, &empty);
+        }
```

Comment:
can replace this and the `cls_cxx_map_remove_key()` call above with `remove_entry()`